### PR TITLE
#80 Add Java support

### LINKED ISSUE #80: Add Java support
Add java support to kube2pulumi

### PR DESCRIPTION
**Pull request by Asamsig**
Fixes #80 

Added Java support, but it seems like the Codegen still needs a few tweaks, currently it generates non-compilable code in some instances.

Issues I saw:
1. Some imports are not unused, and some imports that are needed are not present.
	- `ContainerArgs` does not get imported e.g.
2. Use of multiline string, this is a Java 15 feature, I'm not sure what Java language level the codegen is targeting, seems like the templates set Java 11 ?
3. Use of Map<String, String>, but setting a numeral value.

Figured I'd open this PR even with the above issues present.

EDIT: See my comments for examples of the above issues.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -265,6 +271,60 @@ class MyStack : Stack
...
+    public static void stack(Context ctx) {
+        var fooBarPod = new Pod("fooBarPod", PodArgs.builder()
+                .apiVersion("v1")
+                .kind("Pod")
+                .metadata(ObjectMetaArgs.builder()
+                        .namespace("foo")
+                        .name("bar")
+                        .build())
+                .spec(PodSpecArgs.builder()
+                        .containers(ContainerArgs.builder()
```
Comment (Asamsig) [inline README.md:311]: Missing import for `ContainerArgs`

File: README.md [hunk 2]
```diff
@@ -265,6 +271,60 @@ class MyStack : Stack
...
+                                .name("nginx")
+                                .image("nginx:1.14-alpine")
+                                .resources(ResourceRequirementsArgs.builder()
+                                        .limits(Map.ofEntries(
+                                                Map.entry("memory", "20Mi"),
+                                                Map.entry("cpu", 0.2)
```
Comment (Asamsig) [inline README.md:317]: Number given as key in `Map<String, String>`

File: README.md [hunk 3]
```diff
@@ -265,6 +271,60 @@ class MyStack : Stack
...
+import com.pulumi.kubernetes.core_v1.Pod;
+import com.pulumi.kubernetes.core_v1.PodArgs;
+import com.pulumi.kubernetes.meta_v1.inputs.ObjectMetaArgs;
+import com.pulumi.kubernetes.core_v1.inputs.PodSpecArgs;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Map;
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Paths;
```
Comment (Asamsig) [inline README.md:295]:
```suggestion
import com.pulumi.Context;
import com.pulumi.Pulumi;
import com.pulumi.kubernetes.core_v1.Pod;
import com.pulumi.kubernetes.core_v1.PodArgs;
import com.pulumi.kubernetes.meta_v1.inputs.ObjectMetaArgs;
import com.pulumi.kubernetes.core_v1.inputs.PodSpecArgs;
import java.util.Map;
```

These are the only required imports.

NOTE: I didn't add the missing `ContainerArgs` import, just trying to showcase the added unused imports.

**Comment by Zaid-Ajaj** [comment]
Hi @Asamsig, thanks a lot for the contribution! The code looks good ✅ I will review and test code locally before I can approve. As for the java program-gen issues, these are mostly known but it is better to open issues explicitly for each one of them in [pulumi-java](https://github.com/pulumi/pulumi-java) to track and fix them there.

**Comment by Asamsig** [comment]
@Zaid-Ajaj that is correct, at first I thought it edge cases related to kube2pulumi, but I've reproduced some already in pulumi-java.
I'm working towards contributing fixes there. 👍

**Comment by Zaid-Ajaj** [comment]
@Asamsig that sounds awesome! thanks a lot 🙏

**Comment by Asamsig** [comment]
@Zaid-Ajaj I noticed the tests failed on Github Actions, so I took a look, and it seems that in the MultilineString test it actually expects a file to exist called MultilineString$[ext} (.java), and not a file called expectedMultilineString. I have amended my commit and force pushed, so it should now pass on CI. :)

**Comment by Zaid-Ajaj** [comment]
@Asamsig thanks a lot for investigating and fixing the issue! I am now re-running the CI and will merge with all is green ✅

**Pull request merged**